apps/token-manager/app/src/utils.js

```diff
@@ -0,0 +1,108 @@
+import BN from 'bn.js'
+
+// amount: the total amount (BN.js object)
+// base: the decimals base (BN.js object)
+export function formatBalance(amount, base) {
+  const baseLength = base.toString().length
+
+  let fraction = amount.mod(base).toString()
+  const zeros = '0'.repeat(Math.max(0, baseLength - fraction.length - 1))
+  fraction = `${zeros}${fraction}`
+  const whole = amount.div(base).toString()
+
+  return `${whole}${fraction === '0' ? '' : `.${fraction.slice(0, 2)}`}`
```

Comment:
It is better if we parse the fraction as a number and check if it is 0, otherwise when the fraction has more than 0 zero, the decimals are displayed. Had to change it [here](https://github.com/aragon/radspec/blob/c6cd40356bc1f15c4acdf8c38871de167ea47e56/src/helpers/lib/formatBN.js) in radspec as well. We should probably move this to an utils lib somewhere as I was talking to @bpierre offline today.

<img width="848" alt="screen shot 2018-10-05 at 17 27 10" src="https://user-images.githubusercontent.com/447328/46544456-ef135680-c8c3-11e8-97e2-d4e4b80ae35f.png">

Reply:
Nice UI touch